crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -357,12 +357,12 @@ fn propagate_recursive(
 /// Entities that are visible will be marked as such later this frame
 /// by a [`VisibilitySystems::CheckVisibility`] system.
 fn reset_view_visibility(mut query: Query<&mut ViewVisibility>) {
-    for mut view_visibility in &mut query {
+    query.iter_mut().for_each(|mut view_visibility| {
         // NOTE: We do not use `set_if_neq` here, as we don't care about
         // change detection for view visibility, and adding a branch to every
         // loop iteration would pessimize performance.
-        *view_visibility = ViewVisibility::HIDDEN;
-    }
+        *view_visibility.bypass_change_detection() = ViewVisibility::HIDDEN;
```

Comment:
In general when doing this kind of change my go-to recommendation in review is to leave a comment in code. It's not intuitive that the change yields a 15x speedup, so it's easy for someone else to come back later and change it without realizing they introduce a regression.

Reply:
The comment already mentions why we don't care about change detection, though the point on autovectorization makes sense to me. I'll make a follow-up PR to document this.

Reply:
Can bundle with a later change too. Thanks!

Reply:
What causes this huge speedup? Is it the `for_each()` or the `bypass_change_detection`? I'm not sure I understand

Reply:
Both. See Discord #rendering-dev there's some explanation.